internal/pkg/action/operator_uninstall.go

```diff
@@ -48,9 +48,9 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 	}
 
 	var sub *v1alpha1.Subscription
-	for _, s := range subs.Items {
+	for i, s := range subs.Items {
 		if u.Package == s.Spec.Package {
-			sub = &s
+			sub = &subs.Items[i]
```

Comment:
Does this also solve the problem?
```suggestion
	for i := range subs.Items {
		s := &subs.Items[i]
		if u.Package == s.Spec.Package {
			sub = s	
```

If so, I think I prefer this style better so that we don't mix `subs.Items[i]` and `s` in the same loop, which could be confusing.

Reply:
it does... but we could also skip the index `i` completely with 

```
for _ , s := range subs.Items {
    s := s
		if u.Package == s.Spec.Package {
			sub = &s
```

the main diff is `s := s` copies a value into a new mem location and then is deref with `&s` and `&subs.Items[i]` will deref the value from the original mem address.   (barring compiler / runtime optimizations)

Reply:
Yeah let's do that then. Is this kind of problem common? Or would be make sense to add a comment about this where we have to do it, so someone doesn't come along and think "that looks like a no-op... let's remove this line!"

Reply:
it's very common!    It always shows up for data-driven go tests at a min..  example: https://github.com/kudobuilder/kudo/blob/9db03a9f2be3d7bfe457ebc000881ad4433a27d4/pkg/kudoctl/cmd/diagnostics/print_test.go#L192
also entry #1 of the common mistakes wiki: https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable